src/Npgsql/SingleThreadSynchronizationContext.cs

```diff
@@ -8,7 +9,8 @@ namespace Npgsql;
 sealed class SingleThreadSynchronizationContext : SynchronizationContext, IDisposable
 {
     readonly BlockingCollection<CallbackAndState> _tasks = new();
-    Thread? _thread;
+    volatile Thread? _thread;
+    int _doingWork;
```

Comment:
Not really necessary due to exclusive mutation under CAS and MemoryBarrier (once the error handling is fixed). Adding volatile won't be an issue but it can give a false sense of correctness. Something to say for either choice. In the IOQueue it's absent as far as that means anything.